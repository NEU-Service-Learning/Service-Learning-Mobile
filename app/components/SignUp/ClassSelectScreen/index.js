import React, { Component } from 'react';
import {StyleSheet, View, Text, TextInput, TouchableHighlight } from 'react-native';

import {SearchTable, SearchRow} from '../searchTable'
import SearchBar from './searchBar'

var style = require('../../../Styles/styles');

// Dummy data
var classes = [
  {
    classNum: 'CS 4500',
    name: 'Software dev',
    prof: 'Wientraub',
    projects: ['Mobile Time Tracker', 'Back-end Service-Learning', 'Front-end Service-Learning']
  },
  {
    classNum: 'ARTF 1122',
    name: 'Art and stuff',
    prof: 'Thorne',
    projects: ['Madison High School']
  },
  {
    classNum: 'PT 2568',
    name: 'Physical Therapy and stuff',
    prof: 'Parker',
    projects: ['Hospital Service']
  }
]

export default class ClassSelectScreen extends Component {

  constructor(props) {
    super(props);
    this.state = {searchText: '', selectedClasses: []};
  }

  // Called when the search text changes and updates the state
  handleInput(searchText) {
    this.setState({searchText});
  }

  // Called when a new class is added by clicking on it
  // Appends the class to the list of added classes in the state
  onAddClass(addedClass) {
    const classes = [...this.state.selectedClasses, addedClass]
    this.setState({selectedClasses: classes});
  }

  // Called when a added class is removed
  // Updates the state to remove the given class from the list of added classes
  onRemoveClass(removedClass) {
    addedClasses = this.state.selectedClasses.filter((selectedClass) => {
       return selectedClass != removedClass
    });
    this.setState({selectedClasses: addedClasses});
  }

  // add is a boolean flag if you want the 'add' button/text or the 'remove' button/text
  // onClickFn is a function that represents what happens after the user clicks the action
  // Retuns a function that takes a class and returns a single row on how to render that class
  renderRow(add, onClickFn) {
      return (someClass) => {
        return (
          <SearchRow
              onClicked={onClickFn}
              header={someClass.classNum + " - " + someClass.name}
              subHeader={someClass.prof}
              data={someClass}
              key={someClass.name}
              type={add}/>
            )
        }
    }

  // Used to move to the next screen
  // Passes the list of projects from the classes to the next screen
  navigate() {
    const projects = this.state.selectedClasses.reduce((addedProjects, selectedClass) => {
      return [...addedProjects, ...selectedClass.projects];
    }, []);
    this.props.navigator.push({
      title: 'SelectProject',
      // extras is used to pass data to the next screen.
      // the data gets placed in the props of the Component
      extras: {
        projects: projects
      }
    })
  }

  back() {
    this.props.navigator.pop();
  }

  render() {
    const shownClasses = classes.filter((classData) => {
      var show = this.state.searchText.length != 0 &&
       classData.classNum.toLowerCase().includes(this.state.searchText.toLowerCase()) &&
       !this.state. selectedClasses.includes(classData);
       return show;
    });
    return (
      <View style={StyleSheet.flatten([style.container, style.margin16])}>
        <SearchBar
          filterText={this.state.searchText}
          onChangeText={this.handleInput.bind(this)}/>
        <SearchTable
          data={shownClasses}
          header="Avaliable Classes"
          row={this.renderRow(true, this.onAddClass.bind(this))}
          />
          <SearchTable
            data={this.state.selectedClasses}
            header="Added Classes"
            row={this.renderRow(false, this.onRemoveClass.bind(this))}
            />
          <View style={StyleSheet.flatten([style.nav])} >
            <View style={StyleSheet.flatten([style.back])}>
              <TouchableHighlight style={StyleSheet.flatten([style.button, style.height40])} onPress={() => this.back()}>
                <Text style={StyleSheet.flatten([style.buttonText])}> Back </Text>
              </TouchableHighlight>
            </View>
            <View style={StyleSheet.flatten([style.next])}>
              <TouchableHighlight style={StyleSheet.flatten([style.button, style.height40])} onPress={() => this.navigate()}>
                <Text style={StyleSheet.flatten([style.buttonText])}> Next </Text>
              </TouchableHighlight>
            </View>
          </View>
          </View>
    );
  }
}
