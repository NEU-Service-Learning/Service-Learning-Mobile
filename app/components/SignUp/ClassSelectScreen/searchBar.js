import React, { Component } from 'react';
import {StyleSheet, View, TextInput } from 'react-native';

export default class SearchBar extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return(
      <View style={styles.container}>
        <TextInput
          style={styles.input}
          placeholder="Search for a class here!"
          onChangeText={(text) => this.props.onChangeText(text)}
        />
      </View>
    )
  }

}

const styles = StyleSheet.create({
  container: {
    padding: 10
  },
  input: {
    height: 40
  }
});
