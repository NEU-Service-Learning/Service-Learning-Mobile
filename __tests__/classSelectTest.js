import 'react-native';
import React from 'react';
import ClassSelectScreen from '../app/components/SignUp/ClassSelectScreen/index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <ClassSelectScreen />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
