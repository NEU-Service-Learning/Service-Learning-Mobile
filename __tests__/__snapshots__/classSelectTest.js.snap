exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
      "margin": 16,
    }
  }>
  <View
    style={
      Object {
        "padding": 10,
      }
    }>
    <TextInput
      onChangeText={[Function]}
      placeholder="Search for a class here!"
      style={
        Object {
          "height": 40,
        }
      } />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "teal",
        "borderRadius": 6,
        "flex": 1,
        "marginBottom": 10,
        "padding": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 5,
        }
      }>
      Avaliable Classes
    </Text>
    <ScrollView />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "teal",
        "borderRadius": 6,
        "flex": 1,
        "marginBottom": 10,
        "padding": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 5,
        }
      }>
      Added Classes
    </Text>
    <ScrollView />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "backgroundColor": "blue",
          },
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }>
         Next 
      </Text>
    </View>
  </View>
</View>
`;
